Source/SwiftLintFramework/Extensions/NSRegularExpression+SwiftLint.swift

```diff
@@ -8,13 +8,21 @@
 
 import Foundation
 
+private var regexCache = [String: NSRegularExpression]()
+
 public extension NSRegularExpression {
-    public convenience init(pattern: String) throws {
-        try self.init(pattern: pattern,
+    public static func cached(pattern pattern: String) throws -> NSRegularExpression {
```

Comment:
Yeah, I see now, we can't have this functionality as an initializer. My bad. :+1: